hmt_escrow/utils.py

```diff
@@ -46,3 +46,30 @@ def with_retry(fn, retries=3, delay=5, backoff=2):
         wait_time *= backoff
 
     return False
+
+
+def get_hmt_balance(wallet_addr, token_addr, w3):
+    """ Get hmt balance
+
+    Args:
+        wallet_addr: wallet address
+        token_addr: ERC-20 contract
+        w3: Web3 instance
+
+    Return:
+        Decimal with HMT balance
+    """
+    try:
+        abi = [
+            {
+                "constant": True,
+                "inputs": [{"name": "_owner", "type": "address"}],
+                "name": "balanceOf",
+                "outputs": [{"name": "balance", "type": "uint256"}],
+                "type": "function",
+            }
+        ]
+        contract = w3.eth.contract(abi=abi, address=token_addr)
+        return contract.functions.balanceOf(wallet_addr).call()
+    except Exception as e:
+        logger.warning(f"Can't get HMT Balance: {e}")
```

Comment:
Stupid question? Why not let it fail?

Reply:
got used to adding a try/exception to everything that runs in the path of the request. Not needed here I think. removing it.